### PR TITLE
Added Show Filters option, added Year filter on posts edit page, update for showing Year filter

### PR DESCRIPTION
**Pull request by jonathanpaulbbaroma**
## Description
This is a test task assigned by @stevejburge for issue #1412  which adds the Show Filters options under taxonomies. For adding a filter on the edit Post page. a sample Year filter has been added.

## Benefits
Provide users the option to show or hide filters on the edit posts page.

## Possible drawbacks
n/a

## Applicable issues
#1412 

## Checklist

- [x] I have created a specific branch for this pull request before committing, starting off the current HEAD of `development` branch. 
- [x] I'm submitting to the `development`, feature/hotfix/release branch. (Do not submit to the master branch!)
- [x] This pull request relates to a specific problem (bug or improvement).
- [x] I have mentioned the issue number in the pull request description text.
- [x] All the issues mentioned in this pull request relate to the problem I'm trying to solve.
- [x] The code I'm sending follows the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding style.

### CODE REVIEW

**Review by ojopaul — CHANGES_REQUESTED** [review]
@jonathanpaulbbaroma Check this PR https://github.com/TaxoPress/TaxoPress/pull/1462 for example to see the PR rules.

All the template requirement are supposed to be filled to give details about your pull request.

**Comment by jonathanpaulbbaroma** [comment]
Hi.
May I know if you can provide a video on steps or a guide on how actually to create a pull request?

|  | Best regards,
|    | 
| JP Baroma | Senior Website Developer |
| mobile:  +63 928-1493982 |
| phone:  +63 (052) 826- 1380 |
| ***@***.*** |
| skype:  jpbaroma |
| address: Bacacay, Albay 4509 Philippines |

 |


|
| 
|  |  | 
|  |

 |

 |

 |




 

    On Monday, September 12, 2022 at 04:49:26 PM GMT+8, ojopaul ***@***.***> wrote:  
 
 


@ojopaul requested changes on this pull request.

@jonathanpaulbbaroma Check this PR #1462 for example to see the PR rules.

All the template requirement are supposed to be filled to give details about your pull request.

**Comment by ojopaul** [comment]
Kindly check the sample pull request I linked and edit your pull request to
include all information including description, issue link and mark the
checklist to signify adapting to the requirement.

When creating pull request, the templates are shown to you and you're
required to fill in the information before clicking on Pull request but
since you already did, edit current PR and fill in all data in the
descriptions.

If you don't know how to edit PR, check my comment on (PR #1455)

**Comment by jonathanpaulbbaroma** [comment]
> 

Hi @ojopaul @stevejburge I have already updated the pull request. Looking forward for your feedback

File: inc/functions.inc.php [hunk 1]
```diff
@@ -44,6 +44,85 @@ function st_register_widget() {
...
+    return 'true';
+}
+
+/**
+ * Add custom filter on edit posts page
+ * 
+ * Filter posts by year
+ */
+
+//var_dump(get_option('taxopress_external_taxonomies')['category']['show_filters']);
```
Comment (ojopaul) [inline inc/functions.inc.php:63]: `//var_dump(get_option('taxopress_external_taxonomies')['category']['show_filters']);`

Leaving debugging code in final commit is discouraged. After var_dump or doing any debug, always make sure it's removed before making your commit to development.

So, remove every instance of `var_dump` in your commit

File: inc/functions.inc.php [hunk 2]
```diff
@@ -44,6 +44,85 @@ function st_register_widget() {
...
+    global $wpdb, $table_prefix;
+    
+    $post_type = (isset($_GET['post_type'])) ? $_GET['post_type'] : 'post';
+
+    //only add filter to posts
+    if ($post_type == 'post'){
+        
+        //query database to get a list of years for the specific post type:
+        $values = array();
+        $query_years = $wpdb->get_results("SELECT year(post_date) as year from ".$table_prefix."posts
```
Comment (ojopaul) [inline inc/functions.inc.php:86]: I don't see the reasons to query database in this requirement. I'll suggest reviewing the issue properly to understand the requirement and do it the right way.

For adding taxonomy dropdown, WordPress already provide an API `wp_dropdown_categories` which is all needed to show dropdown of taxonomies filter

**Comment by ojopaul** [comment]
> > 
> 
> Hi @ojopaul @stevejburge I **have already updated** the pull request. Looking forward for your feedback

Hello @jonathanpaulbbaroma You just updated the requirement yesterday and not already updated :)

I'll review the PR now and drop my feedback.

In addition, when you make new changes and believe it's ready for review. click the button shown below so i'll know it's ready for review after last changes request:

re-review

File: inc/taxonomies.php [hunk 3]
```diff
@@ -1645,6 +1670,48 @@ public function taxopress_manage_taxonomies()
 <input type="hidden" name="cpt_tax_status" id="cpt_tax_status"
    value="<?php echo esc_attr($tab); ?>"/>
 </p>
+                        <?php
+                        if($taxonomy_edit){
```
Comment (ojopaul) [inline inc/taxonomies.php:1674]: @jonathanpaulbbaroma This whole section was removed in another PR (https://github.com/TaxoPress/TaxoPress/pull/1455) but it's part of your commit. 

Always make sure you work on updated development branch and when making commit, only commit the changes you made and no addition or subtraction of codes that are not part of the requirement.

File: inc/functions.inc.php [hunk 4]
```diff
@@ -44,6 +44,85 @@ function st_register_widget() {
...
+                where post_type='".$post_type."'
+                group by year(post_date)
+                order by post_date");
+        foreach ($query_years as &$data){
+            $values[$data->year] = $data->year;
+        }
+
+        //give a unique name in the select field
+        ?><select name="tp_fiilter_year">
+<option value="">All years</option>
```
Comment (ojopaul) [inline inc/functions.inc.php:96]: @jonathanpaulbbaroma All strings should be translation ready. Hardcoding text without passing it through translation key will make it impossible for translator to find newly added strings when translating or making update to translation files.

**Review by ojopaul — CHANGES_REQUESTED** [review]
Hello @jonathanpaulbbaroma I've added my feedback on code aspect but in general, the requirement wasn't to add a sample year filter but instead, to allow users to show a taxonomy filter for on Posts screen.

Kindly check the issue requirement https://github.com/TaxoPress/TaxoPress/issues/1412 and see attached image and description for better understanding.

**Comment by jonathanpaulbbaroma** [comment]
Hi @ojopaul @stevejburge Please check the new commits that I had for this PR to resolve the feedback.

File: inc/functions.inc.php [hunk 5]
```diff
@@ -44,6 +44,80 @@ function st_register_widget() {
...
+            $tax_label = $user_taxonomy["label"];
+            
+            $show_filters = tp_disp_boolean($user_taxonomy["show_filters"]);
+
+            if ( $show_filters == 'true' ) {
+                
+                $args = array (
+                    'taxonomy' => $tax_name,
+                    'name'     => 'cat-' . $tax_name,
+                    'show_option_none' => __( 'All ' . $tax_label, 'textdomain' ),
```
Comment (ojopaul) [inline inc/functions.inc.php:86]: This is not TaxoPress translation text domain

File: inc/functions.inc.php [hunk 6]
```diff
@@ -44,6 +44,80 @@ function st_register_widget() {
...
+add_action('restrict_manage_posts', 'taxopress_add_posts_filter');
+add_filter( 'parse_query', 'taxopress_user_filters');
+
+function taxopress_add_posts_filter() {
+
+    global $wpdb, $wp_query, $table_prefix;
+    
+    $post_type = (isset($_GET['post_type'])) ? $_GET['post_type'] : 'post';
+    
+    if ($post_type == 'post'){ //only add filter to posts
```
Comment (ojopaul) [inline inc/functions.inc.php:71]: I already replied your comment on the issue page.

The dropdown should be enabled on the taxonomies post types

File: inc/functions.inc.php [hunk 7]
```diff
@@ -44,6 +44,80 @@ function st_register_widget() {
...
+ * Add filter for custom taxonomies on edit posts page
+ * 
+ */
+
+add_action('restrict_manage_posts', 'taxopress_add_posts_filter');
+add_filter( 'parse_query', 'taxopress_user_filters');
+
+function taxopress_add_posts_filter() {
+
+    global $wpdb, $wp_query, $table_prefix;
```
Comment (ojopaul) [inline inc/functions.inc.php:67]: This global variables were never used on the page and as a result should be removed

**Review by ojopaul — CHANGES_REQUESTED** [review]
@jonathanpaulbbaroma I've updated the feedback based on new commits

**Pull request closed without merging**